client/rest-high-level/build.gradle

```diff
@@ -26,9 +26,9 @@ group = 'org.elasticsearch.client'
 dependencies {
   compile "org.elasticsearch:elasticsearch:${version}"
   compile "org.elasticsearch.client:rest:${version}"
-  compile "org.elasticsearch.plugin:parent-join-client:${version}"
-  compile "org.elasticsearch.plugin:aggs-matrix-stats-client:${version}"
```

Comment:
I agree with @javanna. This functionality is in the ES distribution, therefore the client should have it accessible with no extra work. However, I also see the benefit of keeping the core client lightweight, and having extra functionality added optionally. But I think that could be discussed and debated separately from this PR?